src/main/app/src/store/reducers/oppilaitosSlice.js

```diff
@@ -0,0 +1,129 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { getOppilaitos, getOppilaitosTajonta } from '#/src/api/konfoApi';
+import { Localizer as l } from '#/src/tools/Utils';
+
+const IDLE_STATUS = 'idle';
+const LOADING_STATUS = 'loading';
+
+export const initialState = {
+  status: IDLE_STATUS,
+  oppilaitokset: {},
+  tarjonta: {},
+  tulevaTarjonta: {},
+  page: 1,
+  size: 5,
+  offset: 0,
+  tulevaPage: 1,
+  tulevaSize: 3,
+  tulevaOffset: 0,
+  order: 'asc',
+  oppilaitosError: null,
+};
+
+const oppilaitosSlice = createSlice({
+  name: 'oppilaitos',
+  initialState,
+  reducers: {
+    fetchOppilaitosStart(state) {
+      if (state.status === IDLE_STATUS) {
+        state.status = LOADING_STATUS;
+      }
+    },
+    fetchOppilaitosSuccess(state, { payload }) {
+      if (state.status === LOADING_STATUS) {
+        const { oid, oppilaitosData, tarjonta, tulevaTarjonta } = payload;
+        state.oppilaitokset[oid] = oppilaitosData;
+        state.tarjonta[oid] = tarjonta;
+        state.tulevaTarjonta[oid] = tulevaTarjonta;
```

Comment:
Onko järkeä tallentaa usean `oid`:n juttuja reduxiin tässä? Eihän näitä kai uudelleenkäytetä ilman latausta missään? Jos jostain syystä tulee useita requesteja yhtä aikaa eri oideille, niin sitten voi myös tulla mielenkiintoisia sivuvaikutuksia, koska LOADING_STATUS redux-tila päivitetään vain ensimmäisenä päättyvän requestin osalta.